lib/config/formatter.go

```diff
@@ -323,3 +332,46 @@ func frameToTrace(frame runtime.Frame) trace.Trace {
 		Line: frame.Line,
 	}
 }
+
+func (r knownFormatFieldsMap) has(name string) bool {
+	_, ok := r[name]
+	return ok
+}
+
+func (r knownFormatFieldsMap) names() (result []string) {
+	for k := range r {
+		result = append(result, k)
+	}
+	return result
+}
+
+type knownFormatFieldsMap map[string]struct{}
+
+// KnownFormatFields ...
+var KnownFormatFields = knownFormatFieldsMap{
```

Comment:
nit: you either use `TextFormatterFields` or `KnownFormatFields` and not both - they duplicate each other.